libs/core/tests/unit_tests/prompts/test_prompt.py

```diff
@@ -38,6 +38,135 @@ def test_prompt_from_template() -> None:
+        "properties": {
+            "bar": {"title": "Bar", "type": "string"},
+            "foo": {"title": "Foo", "type": "string"},
+        },
+    }
+
+    # Nested variables.
+    template = "This {{obj.bar}} is a {{obj.foo}} test {{foo}}."
+    prompt = PromptTemplate.from_template(template, template_format="mustache")
+    assert prompt.format(obj={"bar": "foo", "foo": "bar"}, foo="baz") == (
+        "This foo is a bar test baz."
+    )
+    assert prompt.input_variables == ["foo", "obj"]
+    assert prompt.input_schema.schema() == {
+        "title": "PromptInput",
+        "type": "object",
+        "properties": {
+            "foo": {"title": "Foo", "type": "string"},
+            "obj": {"$ref": "#/definitions/obj"},
+        },
+        "definitions": {
+            "obj": {
+                "title": "obj",
+                "type": "object",
+                "properties": {
+                    "foo": {"title": "Foo", "type": "string"},
+                    "bar": {"title": "Bar", "type": "string"},
+                },
+            }
+        },
+    }
+
+    # . variables
+    template = "This {{.}} is a test."
+    prompt = PromptTemplate.from_template(template, template_format="mustache")
+    assert prompt.format(foo="baz") == ("This {'foo': 'baz'} is a test.")
+    assert prompt.input_variables == []
+    assert prompt.input_schema.schema() == {
+        "title": "PromptInput",
+        "type": "object",
+        "properties": {},
+    }
+
+    # section/context variables
+    template = """This{{#foo}}
+        {{bar}}
+    {{/foo}}is a test."""
+    prompt = PromptTemplate.from_template(template, template_format="mustache")
+    assert prompt.format(foo={"bar": "yo"}) == (
+        """This
+        yo
+    is a test."""
+    )
+    assert prompt.input_variables == ["foo"]
+    assert prompt.input_schema.schema() == {
+        "title": "PromptInput",
+        "type": "object",
+        "properties": {"foo": {"$ref": "#/definitions/foo"}},
+        "definitions": {
+            "foo": {
+                "title": "foo",
+                "type": "object",
+                "properties": {"bar": {"title": "Bar", "type": "string"}},
+            }
+        },
+    }
+
+    # section/context variables with repeats
+    template = """This{{#foo}}
+        {{bar}}
+    {{/foo}}is a test."""
+    prompt = PromptTemplate.from_template(template, template_format="mustache")
+    assert prompt.format(foo=[{"bar": "yo"}, {"bar": "hello"}]) == (
+        """This
+        yo
+    
+        hello
+    is a test."""
+    )
+    assert prompt.input_variables == ["foo"]
+    assert prompt.input_schema.schema() == {
+        "title": "PromptInput",
+        "type": "object",
+        "properties": {"foo": {"$ref": "#/definitions/foo"}},
+        "definitions": {
+            "foo": {
+                "title": "foo",
+                "type": "object",
+                "properties": {"bar": {"title": "Bar", "type": "string"}},
```

Comment:
Should be string or array i think? The array is also needed to support inverted sections?

Reply:
cc @nfcampos this might still be an issue